backend/vouch/tests/test_trust_algo.py

```diff
@@ -68,55 +70,77 @@ def setUp(self) -> None:
+        pretrust = np.random.randint(2, size=self._nb_users)
+        relative_pretrust = pretrust/np.sum(pretrust)
+        normalized_vouch_matrix = normalize_vouch_matrix(vouch_matrix, pretrust)
+        relative_posttrust = compute_relative_posttrust(normalized_vouch_matrix, relative_pretrust)
 
         # ensure it sums to 1
-        self.assertAlmostEqual(np.sum(trust_vec), 1)
-        # ensure sum_trusted >= a
-        self.assertTrue(np.sum([trust_vec[i] for i in range(len(trust_vec)) if p[i] > 0]) >= 0.2)
-        # ensure that untrusted users than aren't vouched for don't get trust
-        p[9] = 0
-        C[:, 9] = 0
-        p = p/np.sum(p)
-        C_normalized = normalize_trust_values(C, p)
-        trust_vec = get_trust_vector(C_normalized, p)
-        self.assertAlmostEqual(trust_vec[9], 0)
+        self.assertAlmostEqual(np.sum(relative_posttrust), 1)
+        # ensure sum_trusted >= PRETRUST_BIAS
+        self.assertTrue(
+            np.sum([relative_posttrust[u] for u in range(len(relative_posttrust)) if pretrust[u] > 0]) >= PRETRUST_BIAS
+        )
+        # ensure that all pretrusted users have a positive relative posttrust
+        for u in range(len(relative_posttrust)):
+            if pretrust[u] > 0:
+                self.assertTrue(relative_posttrust[u] > 0)
 
-    def test_rescale(self):
-        trust_vect = np.random.randint(0, 100, 10)
-        trust_vect = trust_vect/np.sum(trust_vect)
-        trust_stat = np.random.randint(0, 2, 10)
-        min_idx = np.argmin([trust_vect[i] if trust_stat[i] ==
-                            1 else 2 for i in range(len(trust_vect))])
-        scale_fac = trust_vect[min_idx]
-        rescaled = rescale(trust_vect, trust_stat)
-        self.assertAlmostEqual(rescaled[min_idx], 1)
-        for i in range(len(trust_stat)):
-            self.assertAlmostEqual(rescaled[i] * scale_fac, trust_vect[i])
+        # ensure that untrusted users than aren't vouched for don't get trust
+        untrusted_user = self._nb_users - 1
+        pretrust[untrusted_user] = 0
+        vouch_matrix[:, untrusted_user] = 0
+        relative_pretrust = pretrust/np.sum(pretrust)
+        normalized_vouch_matrix = normalize_vouch_matrix(vouch_matrix, pretrust)
+        relative_posttrust = compute_relative_posttrust(normalized_vouch_matrix, relative_pretrust)
+        self.assertAlmostEqual(relative_posttrust[untrusted_user], 0)
 
-            if trust_stat[i] > 0:
-                self.assertTrue(rescaled[i] >= 0.999999999)
+    # The voting rights of pretrusted users must be at least MIN_PRETRUST_VOTING_RIGHT
+    # Moreover, unless the voting right is clipped to 1, it must be proportional to posttrust
+    def test_compute_voting_rights(self):
+        pretrust = np.random.randint(2, size=self._nb_users)
+        posttrust = np.random.randint(20, 100, self._nb_users)
+        for u in range(len(pretrust)):
+            if pretrust[u] == 0 and np.random.rand() < 0.5:
+                posttrust[u] = 0
```

Comment:
Depending on your answer to the previous comment, I think these lines could be simplified